src/onemod/backend/local_backend.py

```diff
@@ -62,24 +62,35 @@ def _evaluate_stage(
     method : str, optional
         Name of method to evaluate. Default is 'run'.
     subset_id : int, optional
-        Submodel data subset ID if `model` is a `ModelStage` instance.
-        If None, evaluate all data subsets. Default is None
+        Submodel data subset ID. If None, evaluate all data subsets.
+        Default is None
     param_id : int, optional
-        Submodel parameter set ID if `model` is a `ModelStage` instance.
-        If None, evaluate all parameter sets. Default is None.
+        Submodel parameter set ID. If None, evaluate all parameter sets.
+        Default is None.
+
+    Notes
+    -----
+    If stage uses both `groupby` and `crossby`, both `subset_id` and
+    `param_id` must be passed to evaluate in individual submodels,
+    otherwise all submodels will be evaluated.
 
     """
-    if isinstance(stage, ModelStage):
-        if method == "collect":
-            stage.collect()
-        else:
-            if subset_id is None and param_id is None:
+    if method == "collect":
+        stage.collect()
+    else:
+        stage_method = stage.__getattribute__(method)
+        if stage.has_submodels:
+            subset_passed = stage.groupby is not None and subset_id is not None
+            param_passed = stage.crossby is not None and param_id is not None
```

Comment:
Here I think maybe we should error out if `stage.groupby is None and subset_id is not None` same with the `crossby`

Reply:
Good idea